pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxReceiveOperator.java

```diff
@@ -79,6 +79,12 @@ public MailboxReceiveOperator(MailboxService<Mailbox.MailboxContent> mailboxServ
           singletonInstance = serverInstance;
         }
       }
+
+      // FIXME: there's a bug where singletonInstance may be null in the case of a JOIN where
+      // one side is BROADCAST and the other is SINGLETON (this is the case with nested loop
+      // joins for inequality conditions). This causes NPEs in the logs, but actually works
+      // because the side that hits the NPE doesn't expect to get any data anyway (that's the
+      // side that gets the broadcast from one side but nothing from the SINGLETON)
```

Comment:
can we create an issue and link it here?

Reply:
https://github.com/apache/pinot/issues/9592 - added to comment as well